jsjaws.py

```diff
@@ -160,6 +161,31 @@ def execute(self, request: ServiceRequest) -> None:
                 f.write(file_content)
                 file_path = f.name
 
+        # This is a VBScript method of setting an environment variable:
+        #
+        # var wscript_shell_object = CreateObject("WScript.Shell")
+        # var wscript_shell_object_env = wscript_shell_object.Environment("USER")
+        # wscript_shell_object_env("test") = "Hello World!"
+        #
+        # The above code is also valid in JavaScript when we are not intercepting the
+        # WScript.Shell object. However, since we are doing so, the act of
+        # setting the environment variable using round brackets is not possible and will
+        # result in an "ReferenceError: Invalid left-hand side in assignment"
+        # error.
+        #
+        # Therefore we are going to hunt for instances of this, and replace
+        # it with an accurate JavaScript technique for setting variables.
+        matches = re.findall(VBSCRIPT_ENV_SETTING_REGEX, file_content)
+        if matches:
+            new_content = re.sub(VBSCRIPT_ENV_SETTING_REGEX, b"[\\1] = \\2;", file_content)
```

Comment:
I think it would be cleaner to avoid searching twice. If you want a log per match while using re.sub you can use a function as the replacement that does the logging. Something like:
``` python
def logging_replace(match):
    self.log.debug(f"Replaced VBScript Env variable: ({truncate(group_1)}) = {truncate(group_2)};")
    return f"{match.group(1)} = {match.group(2)};"
```

Reply:
Thank you regex god https://github.com/CybercentreCanada/assemblyline-service-jsjaws/pull/173/commits/77638073bda39394f80349842a99a6853ba4ff31